products/views.py

```diff
@@ -1,3 +1,62 @@
-from django.shortcuts import render
+import datetime
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+
+from dr_martens.checkitem   import CheckItem
+from dr_martens.login_check import login_check
+from products.models        import (Product,
+                                    ProductImage,
+                                    ProductOption)
+
+class ProductDetailView(View):
+    @login_check
+    def get(self, request, *args, **kwargs):
+        try:
+            product           = Product.objects.get(id=kwargs["product_id"])
+            product_name      = product.korean_name
+            product_id        = product.id
+            material          = product.material
+            country_of_origin = product.country_of_origin
+
+            product_options   = product.productoption_set.all()
+            product_images    = product.productimage_set.all()
+            price = product_options[0].price
+    
+            colors = [
+                product_option.color.color 
+                for product_option in product_options
+            ]
+            colors = list(set(colors))
+
+            sizes_stocks = [
+                {
+                    "size":product_option.size.name,
+                    "quantity":product_option.stock
+                }
+                for product_option in product_options
+            ]
+             
+            img_urls = [
+                product_image.url
+                for product_image in product_images
+            ]
```

Comment:
```suggestion
            img_urls = [product_image.url for product_image in product_images]
```
한 줄로 만드는 것이 가독성이 조금 더 좋지 않을까요?

Reply:
리뷰 감사합니다.
리뷰 반영하여 수정하였습니다.